src/schema/v2/partner/createPartnerContactMutation.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  GraphQLBoolean,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+
+// Check whats required and whats optional
+interface Input {
+  partnerID: string
+  name?: string
+  position?: string
+  canContact?: boolean
+  email?: string
+  emailConfirmation?: string
+  phone?: string
+  locationID?: string
+}
+
+const CreatePartnerContactMutationResponseType = new GraphQLObjectType<
+  any,
+  ResolverContext
+>({
+  name: "CreatePartnerContactResponse",
+  fields: () => ({
+    id: { type: new GraphQLNonNull(GraphQLString) },
+    name: { type: GraphQLString },
+    position: { type: GraphQLString },
+    canContact: { type: GraphQLBoolean },
+    email: { type: GraphQLString },
+    phone: { type: GraphQLString },
+    locationID: { type: GraphQLString },
+  }),
+})
+
+export const createPartnerContactMutation = mutationWithClientMutationId<
+  Input,
+  any,
+  ResolverContext
+>({
+  name: "createPartnerContact",
+  description: "Creates a new contact for a partner",
+  inputFields: {
+    partnerID: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "ID of the partner",
+    },
+    name: {
+      type: GraphQLString,
+      description: "Contact's name",
+    },
+    position: {
+      type: GraphQLString,
+      description: "Contact's position at the partner",
+    },
+    canContact: {
+      type: GraphQLBoolean,
+      description:
+        "If true, send all user inquiries and order notifications to this contact.",
+    },
+    email: {
+      type: GraphQLString,
+      description: "Email address of the contact",
+    },
+    // Do we need this? is it persisted in the database? What does volt use to read it
+    emailConfirmation: {
+      type: GraphQLString,
+      description: "Confirmation of the email address",
+    },
```

Comment:
The volt form has this email confirmation field but I dont see it being used anywhere in gravity endpoint, contact creation, or in the database


<img width="1656" alt="Screenshot 2025-03-20 at 5 35 28 PM" src="https://github.com/user-attachments/assets/c07363c2-f5b8-44b2-95c9-918c5c2c6da6" />

Reply:
Actually, now that I say this outloud, it looks like volt does the validation of just checking that the email in the form alone matches the the user input....

See [here](https://github.com/artsy/volt/blob/main/app/views/contacts/_form.html.haml#L30)

Reply:
I'm going to pull this out of the input list ^